src/gleam_stdlib.mjs

```diff
@@ -630,3 +630,22 @@ export function decode_field(value, name) {
     return error();
   }
 }
+
+// Used to test side-effects
+let storage = [];
```

Comment:
Maybe pass atom in so the storage can hold mulitple things:
...for Erlang `ets:new(GleamTestStorageAtom)` which comes from the test module. Then it is the users responsibility to handle that mutable store.
Same for js storage `let storage = {};` and `storage_init(store_key)`?